cmd/tink-cli/cmd/get/get.go

```diff
@@ -0,0 +1,100 @@
+package get
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/jedib0t/go-pretty/table"
+	"github.com/spf13/cobra"
+)
+
+type CmdOpt struct {
+	// Headers is the list of headers you want to print as part of the list
+	Headers []string
+	// RetrieveData reaches out to Tinkerbell and it gets the required data
+	RetrieveData func(context.Context) ([]interface{}, error)
+	// PopulateTable populates a table with the data retrieved with the RetrieveData function.
+	PopulateTable func([]interface{}, table.Writer) error
+
+	// Output specifies the output you want the list of resources printed
+	// out. By default it is table but it can be JSON ar CSV.
+	Output string
+	// NoHeaders does not print the header line
+	NoHeaders bool
+}
+
+const shortDescr = `
+Display one or many resources`
+
+const longDescr = `
+Prints a table contains the most important information about a specific
+resource. You can specify the kind of output you want to receive. It can be
+table or csv or json.
+`
+
+const exampleDescr = `
+# List all hardware in table output format.
+tink hardware get
+
+# List all workflow in csv output format.
+tink template get -o csv
+
+# List a single template in json output format.
+tink workflow get -o json [id]
+`
+
+func NewGetCommand(opt CmdOpt) *cobra.Command {
```

Comment:
I kind of would like to have `NewGetCommand` take in a string that it uses to populate the parent in the example.
```go
func NewGetCommand(opt CmdOpt, parent string)
...
   Example: strings.Replace(exampleDescr, "$parent", parent, -1),
...
)
```
 
This way all the examples make the most amount of sense and don't have example for things we aren't trying to do.